src/app/features/tech-record/components/tech-record-summary/tech-record-summary.component.ts

```diff
@@ -84,16 +84,15 @@ export class TechRecordSummaryComponent implements OnInit {
   }
 
   calculateVehicleModel(): void {
-    this.vehicleTechRecordCalculated = this.isEditable
-      ? { ...cloneDeep(this.vehicleTechRecord), reasonForCreation: '' }
-      : this.vehicleTechRecord;
+    this.vehicleTechRecordCalculated = this.isEditable ? { ...cloneDeep(this.vehicleTechRecord), reasonForCreation: '' } : this.vehicleTechRecord;
 
-    this.store.dispatch(updateEditingTechRecord({techRecord: this.vehicleTechRecordCalculated}));
+    this.store.dispatch(updateEditingTechRecord({ techRecord: this.vehicleTechRecordCalculated }));
   }
 
   handleFormState(event: any): void {
-    this.vehicleTechRecordCalculated = merge(cloneDeep(this.vehicleTechRecordCalculated), event)
-    this.store.dispatch(updateEditingTechRecord({techRecord: this.vehicleTechRecordCalculated}));
+    const weightsValue = this.weights?.form.getCleanValue(this.weights?.form);
+    this.vehicleTechRecordCalculated = merge(cloneDeep(this.vehicleTechRecordCalculated), weightsValue, event);
```

Comment:
You don't need to do this manually. It will duplicate the change. The `event` coming from `app-weights` should contain the changes.
```suggestion
    this.vehicleTechRecordCalculated = merge(cloneDeep(this.vehicleTechRecordCalculated), event)
    this.store.dispatch(updateEditingTechRecord({ techRecord: this.vehicleTechRecordCalculated }));
```